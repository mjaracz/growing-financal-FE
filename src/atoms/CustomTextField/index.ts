export * from './CustomTextField';
export * from './useStyle'