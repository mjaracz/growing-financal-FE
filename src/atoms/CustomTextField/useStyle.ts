import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

export const useStyle = makeStyles((theme: Theme) => 
  createStyles({
    textField: {
      margin: 15,
      width: 180,
      '& > div > input': {
        color: theme.palette.text.secondary
      },
      '& > div > fieldset': {
        borderColor: theme.palette.text.secondary
      },
      '& > div[class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-formControl"] > fieldset': {
        borderColor: 'rgb(215, 146, 247) !important'
      },
      '& > div[class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error Mui-focused Mui-focused MuiInputBase-formControl"] > fieldset': {
        borderColor: 'rgb(215, 146, 247) !important'
      },
      '& > p': {
        color: 'rgb(215, 146, 247) !important'
      },
      '& > label': {
        color: theme.palette.text.secondary
      },
      '& > div > div.MuiAutocomplete-endAdornment > button > span > svg': {
        color: theme.palette.text.secondary
      },
      '& > div > div > button > span > svg': {
        color: theme.palette.text.secondary,
      }
    },
  }))