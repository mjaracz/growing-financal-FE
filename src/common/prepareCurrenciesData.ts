import { CurrenciesResponse } from 'molecules/currencySelect';
import { CurrencyRate } from 'molecules/currencySelect';

export function prepareCurrenciesData(currenciesResponse: CurrenciesResponse): CurrencyRate[] {
  const currenciesKeys: string[] = Object.keys(currenciesResponse.rates);
  const currenciesValues: number[] = Object.values(currenciesResponse.rates);
  const preparedCurrencies = [];
  for (let i = 0; i < currenciesKeys.length; i++) {
    const currencyRate = {}
    currencyRate[currenciesKeys[i]] = currenciesValues[i];
    preparedCurrencies.push(currencyRate)
  }
  return preparedCurrencies;
}