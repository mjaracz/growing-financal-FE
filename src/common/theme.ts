import { createMuiTheme } from '@material-ui/core';

export const theme = createMuiTheme({
  palette: {
    primary: {
      main: 'hsl(193, 95%, 68%)',
      dark: 'hsl(220, 13%, 18%)'
    },
    secondary: {
      main: 'hsl(281, 100%, 80%)'
    },
    background: {
      default: 'hsl(220, 13%, 18%)'
    },
    text: {
      secondary: 'hsl(0, 0%, 100%)',
      hint: 'rgb(215, 146, 247)'
    }  
  }
})