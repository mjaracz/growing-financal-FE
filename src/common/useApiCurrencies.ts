export const useApiCurrencies = () => {
  const baseUrl = 'http://api.exchangeratesapi.io/v1/latest';
  const apiKey = '481b03536817ae9a3877e3026e237f51';

  const getLatestRate = async () => 
    fetch(baseUrl + `?access_key=${apiKey}`)
      .then(res => res.json())
      .then((data) => {
        if (data.statusCode >= 400) {
          throw new Error('User code error 4** status')
        }
        return data;
      })

  return {
    getLatestRate,
    baseUrl,
    apiKey,
  }
}
