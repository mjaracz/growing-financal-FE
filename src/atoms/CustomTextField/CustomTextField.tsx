import { TextField } from '@material-ui/core';
import { useStyle } from './useStyle';

export const CustomTextField = (props) => {
  const styles = useStyle();
  return (
    <TextField
      {...props}
      className={styles.textField}
      variant='outlined'
    />
  )
}