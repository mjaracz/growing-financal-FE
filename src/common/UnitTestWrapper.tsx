import React, { FC } from 'react';
import { ThemeProvider } from '@material-ui/core';
import { theme } from './theme';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from 'redux/store';

export const UnitTestWrapper: FC = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          {children}
        </Provider>
      </ThemeProvider>
    </BrowserRouter>
  )
}