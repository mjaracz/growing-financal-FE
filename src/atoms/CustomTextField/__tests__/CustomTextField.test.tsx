import React from 'react';
import { CustomTextField } from '../CustomTextField';
import { render } from '@testing-library/react';
import { UnitTestWrapper } from 'common/UnitTestWrapper';


describe('<CustomTextField />', () => {
  it('should correct match the snapshot', () => {
    const { baseElement } = render(
      <UnitTestWrapper>
        <CustomTextField />
      </UnitTestWrapper>      
    )
    expect(baseElement).toMatchSnapshot();
  })
})
