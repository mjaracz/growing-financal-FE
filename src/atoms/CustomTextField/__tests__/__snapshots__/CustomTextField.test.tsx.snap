// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomTextField /> should correct match the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiFormControl-root MuiTextField-root makeStyles-textField-1"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-3"
            style="width: 0.01px;"
          >
            <span>
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</body>
`;
